src/Service/TransactionService.php

```diff
@@ -75,14 +76,37 @@ function doTransaction(User $user, ?int $amount, string $comment = null, ?int $q
             throw new TransactionInvalidException('Amount can\'t be positive when sending money or buying an article');
         }
 
-        return $this->entityManager->transactional(function () use ($user, $amount, $comment, $quantity, $articleId, $recipientId) {
+        $senderId = $user->getId();
+        return $this->entityManager->transactional(function () use ($senderId, $amount, $comment, $quantity, $articleId, $recipientId) {
             $transaction = new Transaction();
             $transaction->setUser($user);
             $transaction->setComment($comment);
 
+            $sender = null;
+            $recipient = null;
+            if ($recipientId) {
+                if ($senderId < $recipientId) {
```

Comment:
Hm.... that senderId < recipientId doesnt make much sense, right? :D Why shoild that influence the order of getting the sender and recipient?